lib/pure/strutils.nim

```diff
@@ -189,6 +189,33 @@ proc isUpper*(s: string): bool {.noSideEffect, procvar,
   for c in s:
     result = c.isUpper() and result
 
+proc isTitle*(s: string): bool {.noSideEffect, procvar,
```

Comment:
The only reason to have this for ASCII, in my opinion, is for convenience for ASCII characters. I agree that it's not particularly useful (I have never really needed to check if something is a title), but perhaps someone might find it convenient when working with English titles of books or something. 
In any case, if you think it's better suited in the Unicode module and deleted from strutils, I can do that.